pkg/api/health.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	"time"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/models"
+)
+
+func (hs *HTTPServer) databaseHealthyWithCache() bool {
+	cacheKey := "db-healthy"
+
+	if cached, found := hs.CacheService.Get(cacheKey); found {
+		return cached.(bool)
+	}
+
+	healthy := hs.databaseHealthy()
+
+	hs.CacheService.Set(cacheKey, healthy, time.Second*5)
```

Comment:
```suggestion
	hs.CacheService.Set(cacheKey, healthy, time.Second*10)
```

Reply:
I think the default value can be a bit higher. Looks good otherwise feature-wise.